src/workerd/api/node/buffer.c++

```diff
@@ -876,5 +852,17 @@ bool BufferUtil::isUtf8(kj::Array<kj::byte> buffer) {
   return simdutf::validate_utf8(buffer.asChars().begin(), buffer.size());
 }
 
+kj::Array<kj::byte> BufferUtil::transcode(kj::Array<kj::byte> source, kj::String rawFromEncoding, kj::String rawToEncoding) {
+  auto fromEncoding = getEncoding(rawFromEncoding.asPtr());
+  auto toEncoding = getEncoding(rawToEncoding.asPtr());
+
+  // TODO: Throw appropriate error here.
+  if (!i18n::canBeTranscoded(fromEncoding) || !i18n::canBeTranscoded(toEncoding)) {
+    JSG_FAIL_REQUIRE(Error, "Unable to transcode Buffer");
+  }
+
+  return i18n::transcode(source.asPtr(), fromEncoding, toEncoding);
```

Comment:
If the `i18n::transcode` method here accepts a `kj::ArrayPtr<...>` as the first argument, you should be able to just pass `source` here... the `asPtr()` should be unnecessary... at least I think so :-)

Reply:
nice! definitely works.